addons/trenches/functions/fnc_removeTrench.sqf

```diff
@@ -16,32 +16,38 @@
  * Public: No
  */
 
-params ["_trench", "_unit"];
+params ["_trench", "_unit",["_switchingDigger", false]];
 TRACE_2("removeTrench",_trench,_unit);
 
 private _actualProgress = _trench getVariable [QGVAR(progress), 0];
-if(_actualProgress == 0) exitWith {};
+if (_actualProgress <= 0) exitWith {};
 
 // Mark trench as being worked on
 _trench setVariable [QGVAR(digging), true, true];
+_trench setVariable [QGVAR(diggingType), "DOWN", true];
+private _diggerCount = _trench getVariable [QGVAR(diggerCount), 0];
 
-private _removeTime = getNumber (configFile >> "CfgVehicles" >> (typeof _trench) >> QGVAR(removalDuration));
-private _removeTimeLeft = _removeTime * _actualProgress;
+if (_diggerCount > 0) then {
+   if !(_switchingDigger) then {
+      [_trench, _unit] call FUNC(addDigger);
+   };
+}else{
+   _trench setVariable [QGVAR(diggerCount), 1,true];
+};
+
+private _removeTime = missionNamespace getVariable [getText (configFile >> "CfgVehicles" >> (typeOf _trench) >> QGVAR(removalDuration)), 20];
 
 private _placeData = _trench getVariable [QGVAR(placeData), [[], []]];
-_placeData params ["_basePos", "_vecDirAndUp"];
+_placeData params ["", "_vecDirAndUp"];
 
-private _trenchId = _unit getVariable [QGVAR(isDiggingId), -1];
-if(_trenchId < 0) then {
-    _trenchId = GVAR(trenchId);
-    _unit setVariable [QGVAR(isDiggingId), _trenchId, true];
-    GVAR(trenchId) = GVAR(trenchId) + 1;
+if (count _vecDirAndUp == 0) then {
```

Comment:
isEqualTo []
Though you were using isNil check for the same thing in a different place before. Why are you checking isNil in one place, and empty array in the other? If it's really nil then this will fail, if it's really empty array then the other one will fail

Reply:
https://github.com/acemod/ACE3/pull/6712/files#diff-9d78fd656227f30a1c2be691fd7f13a3R42 Here the isNil one

Reply:
Both are there for a reason :D